### PR TITLE
Update waymo open dataset

### PR DESCRIPTION
**Pull request by sharanramjee**
# Add Dataset

* Dataset Name: Waymo Open Dataset
* Issue Reference: #2096 
  
## Description

Update Waymo Open Dataset to add support for v1.1 and v1.2
Main changes among versions:
1. v1.1: Added camera labels for 900 segments
2. v1.2: Added Test Set with 150 segments, plus 800 segments for domain adaptation across Training, Validation, and Test

  
## Checklist
* [X] Address all TODO's
* [X] Add alphabetized import to subdirectory's `__init__.py`
* [X] Run `download_and_prepare` successfully
* [X] Add [checksums file](https://www.tensorflow.org/datasets/add_dataset#2_run_download_and_prepare_locally)
* [X] Properly cite in `BibTeX` format
* [X] Add passing test(s)
* [X] Add test data
* [X] If using additional dependencies (e.g. `scipy`), use [lazy_imports](https://www.tensorflow.org/datasets/add_dataset#extra_dependencies) (if applicable)
* [X] Add data generation script (if applicable)
* [X] [Lint](https://www.tensorflow.org/datasets/add_dataset#5_check_your_code_style) code

### CODE REVIEW

File: tensorflow_datasets/object_detection/waymo_open_dataset_test.py [hunk 1]
```diff
@@ -28,7 +28,8 @@ class WaymoOpenDatasetTest(testing.DatasetBuilderTestCase):
   DATASET_CLASS = waymo_open_dataset.WaymoOpenDataset
   SPLITS = {
       "train": 1,  # Number of fake train example
-      "validation": 1,  # Number of fake test example
+      "validation": 1,  # Number of fake validation example
+      "test": 1,  # Number of fake test example
```
Comment (vijayphoenix) [inline tensorflow_datasets/object_detection/waymo_open_dataset_test.py:32]: Add new `fake_example` file for the `test` split

File: tensorflow_datasets/object_detection/waymo_open_dataset.py [hunk 2]
```diff
@@ -43,11 +43,6 @@
 WARNING: this dataset requires additional authorization and registration.
 Please look at tfds documentation for accessing GCS, and
 afterwards, please register via https://waymo.com/open/licensing/
-
-This dataset is also available in pre-processed format, making it faster
-to load, if you select the correct data_dir:
-tfds.load('waymo_open_dataset', \
-data_dir='gs://waymo_open_dataset_v_1_0_0_individual_files/tensorflow_datasets')
```
Comment (vijayphoenix) [inline tensorflow_datasets/object_detection/waymo_open_dataset.py:50]: Try to update the documentation

**Comment by acharles7** [comment]
@sharanramjee I think you have to use `BuilderConfig` So, we can have two versions of this dataset i.e `waymo/v1.0` and `waymo/v1.2`

File: tensorflow_datasets/object_detection/waymo_open_dataset_test.py [hunk 3]
```diff
@@ -25,15 +25,19 @@
...
       "train": 1,  # Number of fake train example
-      "validation": 1,  # Number of fake test example
+      "validation": 1,  # Number of fake validation example
   }
 
   def setUp(self):
+    """Set up Waymo Open Dataset tests."""
     super(WaymoOpenDatasetTest, self).setUp()
-    self.builder._CLOUD_BUCKET = self.example_dir
+    self.builder.builder_config.cloud_bucket = self.example_dir
```
Comment (Conchylicultor) [inline tensorflow_datasets/object_detection/waymo_open_dataset_test.py:40]: Where the tests really passing ? This line is ignored as the dataset is  recreated in `builder = self._make_builder(config=config)`, so running test would generate the full dataset.

I'll fix this internally, as I feel you should move on to new tasks.
